app/src/main/java/me/quadphase/qpdex/databaseAccess/PokemonFactory.java

```diff
@@ -31,8 +32,176 @@
+    private final String POKEMON_ABILITIES_TABLE = "pokemon_abilities";
+    private final String POKEMON_CAUGHT_TABLE = "pokemon_caught";
+    private final String POKEMON_EGG_GROUPS_TABLE = "pokemon_eggGroups";
+    private final String POKEMON_EVOLUTIONS_TABLE = "pokemon_evolutions";
+    private final String POKEMON_LOCATIONS_TABLE = "pokemon_locations";
+    private final String POKEMON_NATIONAL_ID_TO_UNIQUE_ID_TABLE = "pokemon_nationalID";
+    private final String POKEMON_SUFFIX_TABLE = "pokemon_suffix";
+    private final String POKEMON_TYPES_TABLE = "pokemon_types";
+    private final String TYPE_EFFECTIVENESS_TABLE = "types_effectiveness";
+    private final String POKEMON_MOVES_TABLE = "pokemon_moves";
+
+    private final String NAME = "name";
+    private final String DESCRIPTION = "description";
+    private final String POWER = "power";
+    private final String ACCURACY = "accuracy";
+    private final String PP = "pp";
+    private final String AFFECTS = "affects";
+    private final String GENERATION_FIRST_APPEARED = "genFirstAppeared";
+    private final String HEIGHT = "height";
+    private final String WEIGHT = "weight";
+    private final String ATTACK = "attack";
+    private final String DEFENCE = "defence";
+    private final String HP = "hp";
+    private final String SPECIAL_ATTACK = "spattack";
+    private final String SPECIAL_DEFENCE = "spdefence";
+    private final String SPEED = "speed";
+    private final String CATCH_RATE = "catchRate";
+    private final String CAUGHT = "caught";
+    private final String CONDITION = "condition";
+    private final String GENDER_RATIO_MALE = "genderRatioMale";
+    private final String HATCH_TIME = "hatchTime";
+    private final String LEARN_METHOD = "learnMethod";
+    private final String SUFFIX = "suffix";
+    private final String EFFECTIVE_LEVEL = "effectiveLevel";
+
+    private final String ABILITIY_ID = "abilityID";
+    private final String CATEGORY_ID = "categoryID";
+    private final String EGG_GROUP_ID = "eggGroupID";
+    private final String GAME_ID = "gameID";
+    private final String GENERATION_ID = "generationID";
+    private final String LOCATION_ID = "locationID";
+    private final String MOVE_ID = "moveID";
+    private final String TYPE_ID = "typeID";
+    private final String PARTY_ID = "partyID";
+    private final String POKEMON_UNIQUE_ID = "pokemonUniqueID";
+    private final String POKEMON_NATIONAL_ID = "pokemonNationalID";
+    private final String FROM_TYPE_ID = "fromTypeID";
+    private final String TO_TYPE_ID = "toTypeID";
+    private final String FROM_POKEMON_ID = "fromPokemonID";
+    private final String TO_POKEMON_ID = "toPokemonID";
+
+    private MinimalPokemon[] allMinimalPokemon;
+    private Pokemon[] detailedPokemonShortList;
+    private Pokemon[] allDetailedPokemon;
+
     private static PokemonFactory instance = null;
 
     private HashMap<Integer,Type> types;
```

Comment:
I mentioned it in one of the commits, but for issue #33, we should switch to a "Type[]" object and have the Type class objects hold their key values. Mostly because it eliminates the additional overhead of the HashMap while keeping O(1) access time.